tests_integration/test_snowpark_function.py

```diff
@@ -0,0 +1,417 @@
+from tests_integration.testing_utils.assertions.test_file_assertions import (
+    assert_that_current_working_directory_contains_only_following_files,
+    assert_that_file_content_is_equal_to_snapshot,
+)
+from tests_integration.testing_utils.assertions.test_result_assertions import (
+    assert_that_result_is_successful_and_has_no_output,
+    assert_that_result_is_successful_and_output_contains,
+    assert_that_result_is_successful,
+    assert_that_result_is_successful_and_json_output_contains_value_at_path,
+)
+from tests_integration.testing_utils.naming_utils import ObjectNameProvider
+from tests_integration.testing_utils.sql_utils import SqlTestHelper
+
+
+@pytest.mark.integration
+def test_snowpark_function_flow(_test_steps):
+    _test_steps.assert_that_no_functions_are_in_snowflake()
+    _test_steps.assert_that_no_files_are_staged_in_snowflake()
+
+    _test_steps.snowpark_function_list_should_return_no_data()
+
+    _test_steps.snowpark_function_init_should_initialize_files_with_default_content()
+    _test_steps.snowpark_function_package_should_zip_files()
+
+    function_name = _test_steps.snowpark_function_create_should_finish_successfully()
+    _test_steps.assert_that_only_these_functions_are_in_snowflake(
+        f"{function_name}() RETURN VARCHAR"
+    )
+    _test_steps.assert_that_only_these_files_are_staged_in_snowflake(
+        f"deployments/{function_name}/app.zip"
+    )
+
+    _test_steps.snowpark_function_list_should_return_list_containing_function_at_the_first_place(
+        function_name=function_name,
+        arguments="()",
+        result_type="VARCHAR",
+    )
+
+    _test_steps.snowpark_function_describe_should_return_function_description(
+        function_name=function_name,
+        arguments="()",
+    )
+
+    _test_steps.snowpark_function_execute_should_return_expected_value(
+        function_name=function_name,
+        arguments="()",
+        expected_value="Hello World!",
+    )
+
+    _test_steps.snowpark_function_update_should_finish_successfully(function_name)
+    _test_steps.assert_that_only_these_functions_are_in_snowflake(
+        f"{function_name}() RETURN NUMBER"
+    )
+    _test_steps.assert_that_only_these_files_are_staged_in_snowflake(
+        f"deployments/{function_name}/app.zip"
+    )
+
+    _test_steps.snowpark_function_list_should_return_list_containing_function_at_the_first_place(
+        function_name=function_name,
+        arguments="()",
+        result_type="NUMBER",
+    )
+
+    _test_steps.snowpark_function_execute_should_return_expected_value(
+        function_name=function_name,
+        arguments="()",
+        expected_value=1,
+    )
+
+    _test_steps.snowpark_function_drop_should_finish_successfully(
+        function_name=function_name,
+        arguments="()",
+    )
+    _test_steps.assert_that_no_functions_are_in_snowflake()
+    # I'm not sure if it is expected behavior to leave staged files belonging to the removed function.
```

Comment:
Will be removed after tests with database anyway

Reply:
This my comment in the code was about the lack of removing files from a stage after removing a function. I'm removing the comment because I know now that the behaviour is correct.